functionary/train/training_utils.py

```diff
@@ -72,114 +121,78 @@ def initialize_tokenizer(
-
-            if label == pred:
-                first_token_correct_count += 1
-                first_token_label_dic[label]["correct"] += 1
-
-        if label != -100:
-            if label == pred:
-                acc_count += 1
-            total_num += 1
-        if label in dic:
-            dic[label]["total"] += 1
-            if label == pred:
-                dic[label]["acc"] += 1
-
-    # Calculate the accuracy of first token of the values of parameters
-    unmasked_labels_preds = train_metrics.extract_unmasked_chunks(
-        label_list, prediction_list
-    )
-    first_token_param_value_total, first_token_param_value_acc = 0, 0
-    for unmasked_labels, pred_result in unmasked_labels_preds:
-        try:
-            indices = train_metrics.extract_indices_of_first_tokens_of_param_values_in_assistant_response(
-                tokenizer, unmasked_labels
-            )
-            for index in indices:
-                first_token_param_value_total += 1
-                if unmasked_labels[index] == pred_result[index]:
-                    first_token_param_value_acc += 1
-        except Exception as e:
-            print_rank0(f"encounter exeption: {str(e)}\nFor unmaksed_labels: {unmasked_labels}")
-
-    # Calculate perplexity
-    loss = eval_preds.predictions[1].tolist()
-    loss = sum(loss) / len(loss)
-    perplexity = math.exp(loss)
+    # Calculating perplexity
+    losses = eval_preds.predictions[1]
+    perplexity = np.exp(np.mean(losses))
+
+    # First token-specific accuracy
+    first_token_mask = (labels[:, :-1] == -100) & valid_mask[:, 1:]
+    first_token_correct = (correct_preds[:, 1:] & first_token_mask).sum()
+    first_token_total = first_token_mask.sum()
 
     metrics = {
         "accuracy": acc_count / total_num,
         "perplexity": perplexity,
-        "accuracy_first_token": first_token_correct_count / first_token_total_count,
-        "total_number_first_token": first_token_total_count,
-        "first_token_param_values": first_token_param_value_acc
-        / first_token_param_value_total,
-        "first_token_param_values_total": first_token_param_value_total,
+        "accuracy_first_token": first_token_correct / max(first_token_total, 1),
     }
 
-    for token_id, stat in sorted(
-        first_token_label_dic.items(), key=lambda x: -x[1]["total"]
-    )[:5]:
-        token = tokenizer.decode([token_id])
-        metrics[f"accuracy_first_token_{token}"] = stat["correct"] / stat["total"]
-        metrics[f"accuracy_first_token_{token}_total"] = stat["total"]
-
-    for token_id in dic:
+    # Token-specific accuracies
+    for token_id, stat in token_stats.items():
         token = id2token[token_id]
-        total_num = dic[token_id]["total"]
-        acc = -1
-        if total_num > 0:
-            acc = dic[token_id]["acc"] / total_num
-        metrics[f"accuracy_{token}"] = acc
-        metrics[f"accuracy_total_num_{token}"] = total_num
```

Comment:
I see some metrics has been removed. Could you kindly explain why these changes has been made?